AWSAppSyncClient/Internal/AWSAppSyncRetryHandler.swift

```diff
@@ -12,12 +12,25 @@ struct AWSAppSyncRetryAdvice {
     let retryInterval: DispatchTimeInterval?
 }
 
+// Current implementation has heavy coupling of retry strategy, appsync config and http transport.
+// The logic of shouldRetryRequest should be decoupled for different retryStrategies.
+// TODO: Implement a protocol which accepts a standardized set of inputs, errorResponse, attemptNumber, etc.
+// and returns if retry should be done and after what duration.
+// We can also extend the `AWSAppSyncRetryStrategy` to accept a `custom` enum type which contains a class
+// implementing above protocol.
 final class AWSAppSyncRetryHandler {
     static let maxWaitMilliseconds = 300 * 1000 // 5 minutes of max retry duration.
-
+    static let maxRetryAttemptsWhenUsingAggresiveMode = 12
```

Comment:
In the interests of making things more self contained & testable, what about having actual implementations separately? Something like `AggressiveRetryStrategy` and `ExponentialBackoffRetryStrategy`, which have only one entry point: `retryDelayInMillseconds(for attemptNumber: Int) -> Int`. All logic and configurations can be localized there, and it keeps this file more readable against a hypothetical future where we add another strategy ("laid back"? "hyper aggressive?" :) )